crates/bevy_math/src/curve/mod.rs

```diff
@@ -220,6 +220,101 @@ pub trait Curve<T> {
         })
     }
 
+    /// Create a new [`Curve`] inverting this curve on the x-axis, producing another curve with
+    /// outputs of the same type, effectively playing backwards starting at `self.domain().end()`
+    /// and transitioning over to `self.domain().start()`. The domain of the new curve is still the
+    /// same.
+    ///
+    /// # Notes
+    ///
+    /// - the domain end of this curve has to be bounded
+    fn reverse(self) -> Result<ReverseCurve<T, Self>, ReverseError>
+    where
+        Self: Sized,
+    {
+        self.domain()
+            .has_finite_end()
+            .then(|| ReverseCurve {
+                curve: self,
+                _phantom: PhantomData,
+            })
+            .ok_or(ReverseError::SourceDomainEndInfinite)
+    }
+
+    /// Create a new [`Curve`] repeating this curve either `n` times or indefinitely, producing
+    /// another curve with outputs of the same type. The domain of the new curve will be bigger by
+    /// a factor of `n` or unbounded based on the [`RepeatMode`].
+    ///
+    /// # Notes
+    ///
+    /// - this doesn't guarantee a smooth transition from one occurrence of the curve to its next
+    ///   iteration. The curve will make a jump if `self.domain().start() != self.domain().end()`!
+    /// - for `RepeatMode::Fixed(0)` the output of this adaptor is basically identical to the previous curve
+    /// - the domain of this curve has to be bounded
+    /// - the value at the transitioning points (`domain.end() * n` for `n >= 1`) in the results is the
+    ///   value at `domain.start()` in the original curve
+    fn repeat(self, mode: RepeatMode) -> Result<RepeatCurve<T, Self>, RepeatError>
```

Comment:
I feel like it might just be more ergonomic to break this into two methods instead of using one method with an `enum` parameter:
- A method that takes a curve and make it repeat forever (perhaps to negative infinity as well)
- A method which `chains` a curve with itself `n` times